#933 added unit tests for ItineraryVolunteerListUpdatedHandler

adam-gligor opened the pull request:
Hello, Adam here. 

I heard of you guys on the dotnetrocks podcast and that you are accepting contributions.  
I wrote the unit tests requested in #933. Let me know if you expect further changes to this. 

thanks

tonysurma commented:
Thanks @adam-gligor for contributing.  pinging @mgmccarthy and @MisterJames to give it a quick review and offer any guidance or feedback before we merge.

Thanks again

mgmccarthy commented:
@tonysurma, I'll take a look sometime today or later tonight

mgmccarthy commented on AllReadyApp/Web-App/AllReady.UnitTest/Features/Notifications/IntineraryVolunteerListUpdatedHandlerTests.cs:
```diff
@@ -0,0 +1,121 @@
+
+namespace AllReady.UnitTest.Features.Notifications
+{
+	public class IntineraryVolunteerListUpdatedHandlerTests : InMemoryContextTest
```
rename to `IntineraryVolunteerListUpdatedHandlerShould`

mgmccarthy commented on AllReadyApp/Web-App/AllReady.UnitTest/Features/Notifications/IntineraryVolunteerListUpdatedHandlerTests.cs:
```diff
@@ -0,0 +1,121 @@
+		}
+
+		[Fact]
+		public async Task ShouldDispatchNotificationToThePreferredEmailAddressAndPhoneNumber()
```
Remove the work "Should" from the front of each unit test method name

mgmccarthy commented:
@adam-gligor nice work! 

Can you make `ShouldDispatchVolunteerAssignedNotification` and `ShouldDispatchVolunteerUnAssignedNotification` test the explicit messages being generated by the handler? Right now, looks like you're making sure the items that change (assigned/unassigned) are being assigned correctly.

Also, things like values for `{volunteerDashboardLink}`, `{itineraryDate}` and `{toFrom}` should be tested as well as they relate to the different type of message strings being built.

You might also want to add a test for `NotifyVolunteersCommand` NOT being sent, which should happen in this conditional was met:
```
if (string.IsNullOrWhiteSpace(emailAddress) && string.IsNullOrWhiteSpace(phoneNumber))
    return;
```

Thanks!

adam-gligor commented:
Thanks for the quick feedback.

I made the following changes as you suggested: 

-  Changed the class and method names
-  Updated the DispatchVolunteerAssignedNotification and DispatchVolunteerUnAssignedNotification to test the full content of the message
- Added a new test NotDispatchNotificationWhenNoContactInformation

mgmccarthy commented:
@adam-gligor , I'll try to get to this review by the end of the weekend, no guarantees though :wink:  Most likely, I'll be taking a look Monday. Thanks for the quick turnaround.

MisterJames commented:
Looks good @adam-gligor! Thanks for the original review @mgmccarthy. Looks like everything is in place. :shipit:
